c2corg_ui/static/js/documentediting.js

```diff
@@ -354,5 +382,204 @@ app.DocumentEditingController.prototype.getCoordinatesFromPoint_ = function(
+    if (direction === 'forwards') {
+      // animate till the end (120%)
+      if (step === that.max_steps) {
+        if (will_be >= 120) {
+          clearTimeout(timeout);
+        } else {
+          will_be++;
+        }
+        // animate to the next position ->
+      } else {
+        if (will_be >= next_position) {
+          clearTimeout(timeout);
+        } else {
+          will_be++;
+        }
+      }
+      // if the direction is backwards, animate bar to the left <-
+    } else {
+      if (will_be >= stopBack) {
+        next_position--;
+        will_be = next_position;
+      } else {
+        clearTimeout(timeout);
+      }
+    }
+
+    $('.progress-bar-edit')
+            .css({'background-image': '-webkit-linear-gradient(left, ' + green + ' 0,' + green + ' ' + will_be + '%,' + gray + ' ' + (will_be + 5) + '%,' + gray + ' 0%)'})
+            .css({'background-image': '-o-linear-gradient(left, ' + green + ' 0,' + green + ' ' + will_be + '%,' + gray + ' ' + (will_be + 5) + '%,' + gray + ' 0%)'})
+            .css({'background-image': '-moz-linear-gradient(left, ' + green + ' 0,' + green + ' ' + will_be + '%,' + gray + ' ' + (will_be + 5) + '%,' + gray + ' 0%)'})
+            .css({'background-image': '-ms-linear-gradient(left, ' + green + ' 0,' + green + ' ' + will_be + '%,' + gray + ' ' + (will_be + 5) + '%,' + gray + ' 0%)'})
+            .css({'background-image': 'linear-gradient(left, ' + green + ' 0,' + green + ' ' + will_be + '%,' + gray + ' ' + (will_be + 5) + '%,' + gray + ' 0%)'});
+  }, 10);
+};
+
+/**
+ * Update steps
+ * @export
+ */
+
+app.DocumentEditingController.prototype.updateMaxSteps = function() {
+  var that = this;
+  setTimeout(function() {
+    that.max_steps = $('.step').length;
+    that.scope_.$apply();
+  }, 200)
+}
+
+
+/**
+ * Update arrays and creates one, if not existing
+ * form : model[property] = value, event for finding and (un)checking the checkbox
+ * @param {Object} model
+ * @param {string} property
+ * @param {string} value
+ * @param {Event} event
+ * @export
+ */
+
+app.DocumentEditingController.prototype.pushToArray = function(model, property, value, event) {
+  var checkbox = $(event.currentTarget).find('input');
+  if (!model[property]) {
+    model[property] = [];
+  }
+  if (model[property].indexOf(value) === -1) {
+    model[property].push(value);
+    checkbox.prop('checked', true);
+  } else {
+    model[property].splice(model[property].indexOf(value), 1);
+    checkbox.prop('checked', false);
+  }
+}
+
+
+/**
+ * Update arrays and creates one, if not existing
+ * form : model[property] = value, event for finding and (un)checking the checkbox
+ * @param {Object} model
+ * @export
+ */
+
+app.DocumentEditingController.prototype.setOrientation = function(orientation, waypoint) {
+
+  if (!waypoint['orientation']) {
+    waypoint.orientation = '';
+  }
+
+  waypoint.orientation = orientation;
+
+  $('.circle-selected').removeClass('circle-selected');
```

Comment:
Won't such code be needed in other directives? For instance for filtering lists of documents?
If yes, shouldn't you move the code to https://github.com/c2corg/v6_ui/blob/master/c2corg_ui/static/js/utils.js or some other dedicated files?